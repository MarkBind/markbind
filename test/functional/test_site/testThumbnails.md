**Thumbnail test**

**Images**

**Square thumb, size = 100 (default)**
<thumbnail src="http://placehold.it/800x600" />

**Square thumb with alt="Test"**
<thumbnail alt="Test" src="asdf" />

**Square thumb, size = 200**
<thumbnail src="http://placehold.it/800x600" size='200' />

**Circle thumbs, size = 100**
<thumbnail src="http://placehold.it/800x600" circle />
<thumbnail src="http://placehold.it/100x300" circle />

**An actual image**
<thumbnail circle size=300 src="images/deer.jpg" />
<pic src="images/deer.jpg" height=300 />

---

**Broken link, still takes up the space it's supposed to**

<thumbnail src="/broken_link" border/>
<thumbnail src="http://placehold.it/300x100"/>

---

**Borders**

Thumbnail | Attribute | Description
-----|-----|---------
<thumbnail src="http://placehold.it/300x100" />| |no border
<thumbnail border='3px solid red' src="http://placehold.it/300x100" />|`border='3px solid red'`|3 pixel thick solid red line
<thumbnail border='4px dotted blue' src="http://placehold.it/300x100" />|`4px dotted blue`|4 pixel thick dotted blue line

---

**Background**

**Valid CSS properties**
<thumbnail background="lightblue" circle />
<thumbnail background="#aaf" circle />
<thumbnail background="rgba(120, 120, 200, 0.4)" circle />

**Invalid CSS properties**
<thumbnail background="asdf" border="1px solid black" circle />

**Empty**
<thumbnail background="" border="1px solid black" circle />

---

**Text/Emojis/Icons**

**Text**
<thumbnail text="test" border='1px solid black' circle />

**Markdown**
<thumbnail text="_test_" border='1px solid black' circle />
<thumbnail text="**test**" border='1px solid black' circle />

**Emojis**
<thumbnail text=":x:" border='1px solid black' circle />
<thumbnail text=":eggplant:" border='1px solid black' circle />

**Icons**
<thumbnail text=":fab-github:" border='1px solid black' circle />
<thumbnail text=":glyphicon-book:" background="#666" font-color="#eee" circle />

---

**font-size/font-color**

**20px font-size**
<thumbnail text="test" border='1px solid black' font-size=20 circle />

**40px font-size**
<thumbnail text="test" border='1px solid black' font-size=40 circle />

**font-color**
<thumbnail text="blue" border='1px solid black' font-color="#669" circle />
<thumbnail text="maroon" border='1px solid black' font-color="maroon" font-size="25" circle />
