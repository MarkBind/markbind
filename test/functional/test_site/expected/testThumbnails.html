<!DOCTYPE html>
<html>

<head>
  <meta name="default-head-top">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="MarkBind 2.13.1">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Thumbnails Test</title>
  <link rel="stylesheet" href="markbind/css/bootstrap.min.css">
  <link rel="stylesheet" href="markbind/css/bootstrap-vue.min.css">
  <link rel="stylesheet" href="markbind/fontawesome/css/all.min.css">
  <link rel="stylesheet" href="markbind/glyphicons/css/bootstrap-glyphicons.min.css">
  <link rel="stylesheet" href="markbind/css/octicons.css">
  <link rel="stylesheet" href="markbind/css/github.min.css">
  <link rel="stylesheet" href="markbind/css/markbind.css">
  <link rel="stylesheet" href="STYLESHEET_LINK">
  <link rel="stylesheet" href="markbind/layouts/default/styles.css">
  <meta name="default-head-bottom">
  <link rel="icon" href="/test_site/favicon.png">
</head>

<body>
  <div id="app">
    <div id="flex-body">
      <div id="content-wrapper">
        <p><strong>Thumbnail test</strong></p>
        <p><strong>Images</strong></p>
        <p><strong>Square thumb, size = 100 (default)</strong>
          <thumbnail src="http://placehold.it/800x600"></thumbnail>
        </p>
        <p><strong>Square thumb with alt=&quot;Test&quot;</strong>
          <thumbnail alt="Test" src="/test_site/asdf"></thumbnail>
        </p>
        <p><strong>Square thumb, size = 200</strong>
          <thumbnail src="http://placehold.it/800x600" size="200"></thumbnail>
        </p>
        <p><strong>Circle thumbs, size = 100</strong>
          <thumbnail src="http://placehold.it/800x600" circle=""></thumbnail>
          <thumbnail src="http://placehold.it/100x300" circle=""></thumbnail>
        </p>
        <p><strong>An actual image</strong>
          <thumbnail circle="" size="300" src="/test_site/images/deer.jpg"></thumbnail>
          <pic src="/test_site/images/deer.jpg" height="300"></pic>
        </p>
        <hr>
        <p><strong>Broken link, still takes up the space it's supposed to</strong></p>
        <p>
          <thumbnail src="/broken_link" border=""></thumbnail>
          <thumbnail src="http://placehold.it/300x100"></thumbnail>
        </p>
        <hr>
        <p><strong>Borders</strong></p>
        <div class="table-responsive">
          <table class="markbind-table table table-bordered table-striped">
            <thead>
              <tr>
                <th>Thumbnail</th>
                <th>Attribute</th>
                <th>Description</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  <thumbnail src="http://placehold.it/300x100"></thumbnail>
                </td>
                <td></td>
                <td>no border</td>
              </tr>
              <tr>
                <td>
                  <thumbnail border="3px solid red" src="http://placehold.it/300x100"></thumbnail>
                </td>
                <td><code>border='3px solid red'</code></td>
                <td>3 pixel thick solid red line</td>
              </tr>
              <tr>
                <td>
                  <thumbnail border="4px dotted blue" src="http://placehold.it/300x100"></thumbnail>
                </td>
                <td><code>4px dotted blue</code></td>
                <td>4 pixel thick dotted blue line</td>
              </tr>
            </tbody>
          </table>
        </div>
        <hr>
        <p><strong>Background</strong></p>
        <p><strong>Valid CSS properties</strong>
          <thumbnail background="lightblue" circle=""></thumbnail>
          <thumbnail background="#aaf" circle=""></thumbnail>
          <thumbnail background="rgba(120, 120, 200, 0.4)" circle=""></thumbnail>
        </p>
        <p><strong>Invalid CSS properties</strong>
          <thumbnail background="asdf" border="1px solid black" circle=""></thumbnail>
        </p>
        <p><strong>Empty</strong>
          <thumbnail background="" border="1px solid black" circle=""></thumbnail>
        </p>
        <hr>
        <p><strong>Text/Emojis/Icons</strong></p>
        <p><strong>Text</strong>
          <thumbnail text="test" border="1px solid black" circle="">test</thumbnail>
        </p>
        <p><strong>Markdown</strong>
          <thumbnail text="_test_" border="1px solid black" circle=""><em>test</em></thumbnail>
          <thumbnail text="**test**" border="1px solid black" circle=""><strong>test</strong></thumbnail>
        </p>
        <p><strong>Emojis</strong>
          <thumbnail text=":x:" border="1px solid black" circle="">‚ùå</thumbnail>
          <thumbnail text=":eggplant:" border="1px solid black" circle="">üçÜ</thumbnail>
        </p>
        <p><strong>Icons</strong>
          <thumbnail text=":fab-github:" border="1px solid black" circle=""><span aria-hidden="true" class="fab fa-github"></span></thumbnail>
          <thumbnail text=":glyphicon-book:" background="#666" font-color="#eee" circle=""><span aria-hidden="true" class="glyphicon glyphicon-book"></span></thumbnail>
        </p>
        <hr>
        <p><strong>font-size/font-color</strong></p>
        <p><strong>20px font-size</strong>
          <thumbnail text="test" border="1px solid black" font-size="20" circle="">test</thumbnail>
        </p>
        <p><strong>40px font-size</strong>
          <thumbnail text="test" border="1px solid black" font-size="40" circle="">test</thumbnail>
        </p>
        <p><strong>font-color</strong>
          <thumbnail text="blue" border="1px solid black" font-color="#669" circle="">blue</thumbnail>
          <thumbnail text="maroon" border="1px solid black" font-color="maroon" font-size="25" circle="">maroon</thumbnail>
        </p>
      </div>
    </div>
    <footer>
      <div class="text-center">
        Default footer
      </div>
    </footer>
  </div>
</body>
<script src="markbind/js/vue.min.js"></script>
<script src="markbind/js/vue-strap.min.js"></script>
<script src="markbind/js/bootstrap-utility.min.js"></script>
<script src="markbind/js/polyfill.min.js"></script>
<script src="markbind/js/bootstrap-vue.min.js"></script>
<script>
  const baseUrl = '/test_site'
  const enableSearch = true
</script>
<script src="markbind/js/setup.js"></script>
<script src="SCRIPT_LINK"></script>
<script>
  alert("hello")
</script>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=TRACKING-ID"></script>
<script>
  window.dataLayer = window.dataLayer || [];

  function gtag() {
    dataLayer.push(arguments);
  }
  gtag('js', new Date());

  gtag('config', 'TRACKING-ID');
</script>
<script src="markbind/layouts/default/scripts.js"></script>

</html>